src/var.rs

```diff
@@ -24,17 +24,17 @@ impl<'de, 'a, R: 'a + Read> de::EnumAccess<'de> for EnumAccess<'a, R> {
     fn variant_seed<V: DeserializeSeed<'de>>(
         self,
         seed: V,
-    ) -> Result<(V::Value, VariantAccess<'a, R>), Error> {
-        let name = expect!(
+    ) -> Result<(V::Value, VariantAccess<'a, R>)> {
+        let name: Result<V::Value> = expect!(
             self.de.peek()?,
 
             &XmlEvent::Characters(ref name) |
             &XmlEvent::StartElement { name: OwnedName { local_name: ref name, .. }, .. } => {
                 seed.deserialize(name.as_str().into_deserializer())
             }
-        )?;
+        );
         self.de.set_map_value();
-        Ok((name, VariantAccess::new(self.de)))
+        Ok((name?, VariantAccess::new(self.de)))
```

Comment:
@farodin91 For example, here - what is the reason bailout is now done after `set_map_value` and not before it as before?